bpf/bpf_syscall.go

```diff
@@ -71,7 +71,7 @@ func GetMapFDByID(mapID int) (MapFD, error) {
 const defaultLogSize = 1024 * 1024
 const maxLogSize = 128 * 1024 * 1024
 
-func LoadBPFProgramFromInsns(insns asm.Insns, license string) (fd ProgFD, err error) {
+func LoadBPFProgramFromInsns(insns asm.Insns, license string, forXDP bool) (fd ProgFD, err error) {
```

Comment:
Not sure I follow.  Why would an enum be awkward for bpf_syscall_stub.go?

Reply:
Only because you's need to define your own enum to mirror the constants in the `unix` package (because the unix package doesn't compile on windows).

Reply:
Oh, I was thinking we could have an independent golang enum, and then
```
	switch progType {
	case XDP:
		C.bpf_attr_setup_load_prog(bpfAttr, unix.BPF_PROG_TYPE_XDP, C.uint(len(insns)), cInsnBytes, cLicense, (C.uint)(logLevel), (C.uint)(logSize), logBuf)
	case TC:
		C.bpf_attr_setup_load_prog(bpfAttr, unix.BPF_PROG_TYPE_SCHED_CLS, C.uint(len(insns)), cInsnBytes, cLicense, (C.uint)(logLevel), (C.uint)(logSize), logBuf)
	}
```
instead of
```
	if forXDP {
		C.bpf_attr_setup_load_prog(bpfAttr, unix.BPF_PROG_TYPE_XDP, C.uint(len(insns)), cInsnBytes, cLicense, (C.uint)(logLevel), (C.uint)(logSize), logBuf)
	} else {
		C.bpf_attr_setup_load_prog(bpfAttr, unix.BPF_PROG_TYPE_SCHED_CLS, C.uint(len(insns)), cInsnBytes, cLicense, (C.uint)(logLevel), (C.uint)(logSize), logBuf)
	}
```
It would make the calling code a little more readable, but maybe isn't worth it - WDYT?